src/CMakeLists.txt

```diff
@@ -307,6 +307,18 @@ ENDIF()
 IF(REGEX STREQUAL "regcomp_l")
 	ADD_FEATURE_INFO(regex ON "using system regcomp_l")
 	SET(GIT_REGEX_REGCOMP_L 1)
+ELSEIF(REGEX STREQUAL "pcre2")
```

Comment:
Any reason why we shouldn't auto-detect the PCRE2 library `IF(REGEX STREQUAL "")`?

Reply:
Hmm, I tried to mention this in a comment, but it looks like it got buried.  (It is mentioned in the commit that introduces PCRE2.)

Basically, the POSIX compatibility layer for PCRE2 is completely broken on Ubuntu and Debian.  If you look at PCRE, you'll see that it has symbols of pcre_regcomp, and then `pcreposix.h` sets up that indirection as `#define regcomp pcre_regcomp`.  This avoids conflicting with the regex functions that are in libc.

The PCRE2 packages only do half of this, they have symbols for `PCRE2regcomp` but they forgot to set up the indirection in the header files.  (They just declare `regcomp`.)  So you end up using the regular libc regex bits.  Worse, you use the pcre2 header files and their `#define`s for constants, so you end up passing the wrong constants to libc, and things go very wrong.

So we allow people to set PCRE2 manually, in case they have a working installation, but we don't autodetect it to avoid breaking Debian/Ubuntu users.

Reply:
Hm. Couldn't we just use CMake to detect whether the system PCRE2 is broken or not with `CHECK_FUNCTION_EXISTS`?

Reply:
Hmm, maybe yes, but since they haven't fixed it yet, I don't even have a test case that I can verify that we're getting it correct (yet).

My preference would be to hold off until a (shipping) version of Debian/Ubuntu has this fixed and we can test it...

Reply:
Fair enough. By holding off you probably just refer to the proper detection of non-broken PCRE2, right?

Reply:
Yeah, keep it as-is where people can explicitly opt-in to PCRE2 support but not enable PCRE2 by default (yet).

Reply:
For the record, pcre2 has added compatibility aliases (`#define regexec  pcre2_regexec`) in 10.33 (in [r1064](https://vcs.pcre.org/pcre2?view=revision&revision=1064)).